ml-agents/mlagents/trainers/sac/trainer.py

```diff
@@ -76,12 +76,19 @@ def __init__(
 
         self.checkpoint_replay_buffer = self.hyperparameters.save_replay_buffer
 
-    def save_model(self, name_behavior_id: str) -> None:
+    def _checkpoint(self) -> None:
         """
-        Saves the model. Overrides the default save_model since we want to save
-        the replay buffer as well.
+        Overrides the default to save the replay buffer.
         """
-        self.policy.save_model(self.get_step)
+        super()._checkpoint()
+        if self.checkpoint_replay_buffer:
+            self.save_replay_buffer()
+
+    def save_model(self) -> None:
```

Comment:
save_model and _checkpoint have the same docstring. I doubt they do the same thing

Reply:
Could you please clarify what you meant here, I am not sure I understand the feedback. Is it that they are implementing the same thing?

Reply:
@sankalp04 just that the docstring describing the functions and their params are the same (this was my mistake), and should say what the respective functions do.

Reply:
I think one is supposed to be called by the other. Maybe only one of them needs to be public ?

Reply:
could you clarify @vincentpierre?  I believe `super().save_model()` should result in the `_checkpoint` method being called

Reply:
Ah, my bad. Sorry, I forgot to look at the base class.